src/config/project.rs

```diff
@@ -180,17 +180,26 @@ pub struct ProjectConfig {
 }
 
 impl ProjectConfig {
-    fn parse(dir: &Utf8Path, metadata: &serde_json::Value) -> Result<Self> {
+    fn parse(dir: &Utf8Path, metadata: &serde_json::Value, cargo_metadata: &Metadata) -> Result<Self> {
         let mut conf: ProjectConfig = serde_json::from_value(metadata.clone())?;
         conf.config_dir = dir.to_path_buf();
         let dotenvs = load_dotenvs(dir)?;
         overlay_env(&mut conf, dotenvs)?;
-        if conf.site_root == "/" || conf.site_root == "." {
+        if conf.site_root == "/" || conf.site_root == "." || conf.site_root == "CARGO_TARGET_DIR" {
             bail!(
                 "site-root cannot be '{}'. All the content is erased when building the site.",
                 conf.site_root
             );
         }
+        if conf.site_root.starts_with("CARGO_TARGET_DIR") {
+            conf.site_root = {
+                let mut path = cargo_metadata.target_directory.clone();
+                //unwrap should be safe because we just checked
+                let sub = conf.site_root.unbase("CARGO_TARGET_DIR".into()).unwrap();
```

Comment:
Add synonym. Also, extract to a constant? More than a few places the env var name is used by now :)